e2e-test/java/io/cdap/plugin/slt/stepsdesign/StepDefinitionSLT.java

```diff
@@ -0,0 +1,471 @@
+  public void user_fetches_the_record_count_of_table_something_from_sap(String table)
+    throws JCoException {
+    sapProps = SAPProperties.getDefault(connection);
+    errorCapture = new ErrorCapture(exceptionUtils);
+    sapAdapterImpl = new SAPAdapterImpl(errorCapture, connection);
+    Map opProps = new HashMap<>();
+    opProps.put("RFC", "ZFM_TABLE_COUNT");
+    opProps.put("autoCommit", "true");
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      ObjectNode objectNode = mapper.createObjectNode();
+      objectNode.put("IM_TABLE", table.toUpperCase());
+      JsonNode response = sapAdapterImpl.executeRFC(objectNode.toString(), opProps, "", "");
+      sapRecordsCount = Integer.parseInt(response.get("EX_COUNT").toString().replaceAll("\"", ""));
+      BeforeActions.scenario.write("Total record count : " + sapRecordsCount);
+    } catch (Exception e) {
+      throw SystemException.throwException(e.getMessage(), e);
+    }
+  }
+
+  @Given("Create new MTID on {string} SAP")
+  public void user_creates_mtid_on_sap(String mtIdDes)
+    throws JCoException, InterruptedException {
+    if (mtId == null) {
+      sapProps = SAPProperties.getDefault(connection);
+      errorCapture = new ErrorCapture(exceptionUtils);
+      sapAdapterImpl = new SAPAdapterImpl(errorCapture, connection);
+      Map opProps = new HashMap<>();
+      opProps.put("RFC", "ZRFM_IUUC_CREATE_CONFIG");
+      opProps.put("autoCommit", "true");
+      try {
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode objectNode = mapper.createObjectNode();
+        objectNode.put("IV_SCHEMA_NAME", mtIdDes);
+        objectNode.put("IV_SCHEMA_DESCR", mtIdDes);
+        objectNode.put("IV_APPLICATION", "ZGOOGLE_CDF");
+        objectNode.put("IV_REPLAY_JOBS", "009");
+        objectNode.put("IV_INIT_LOAD_JOBS", "009");
+        objectNode.put("IV_REPL_MODE", "1");
+        objectNode.put("IV_SOURCE_SYSTEM_TYPE", "SAP");
+        objectNode.put("IV_SOURCE_RFCDEST", "CQ1");
+        objectNode.put("IV_TARGET_RFCDEST", "NONE");
+        objectNode.put("IV_HANA_INSTANCE", "0");
+        objectNode.put("IV_HANA_PROXYPORT", "000");
+        objectNode.put("IV_RFC_CLIENT_ONLY", "X");
+        objectNode.put("IV_REUSE_SENDER", "X");
+        objectNode.put("IV_IL_SCENARIOIV_IL_SCENARIO", "1");
+        objectNode.put("IV_NUM_JOBS_ACPCALC", "000");
+        JsonNode response = sapAdapterImpl.executeRFC(objectNode.toString(), opProps, "", "");
+        mtId = response.get("EV_MT_ID").asText();
+        guid = response.get("EV_CONFIG_GUID").asText();
+        System.out.println("MT id : " + mtId + " guid : " + guid);
+      } catch (Exception e) {
+        throw SystemException.throwException(e.getMessage(), e);
+      }
+      Thread.sleep(6000); //sleep required to wait for async operation to be executed on SAP
+    }
+  }
+
+  @When("Validate that MTID can be Updated using CDF_R_SLT_SETTINGS program")
+  public void user_updates_mtid_in_program()
```

Comment:
Check all step definition's method names. Should align with step description.
Applies to all